test/local/ip_profiling.js

```diff
@@ -0,0 +1,283 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var sinon = require('sinon')
+
+const assert = require('insist')
+var mocks = require('../mocks')
+var getRoute = require('../routes_helpers').getRoute
+var proxyquire = require('proxyquire')
+
+var P = require('../../lib/promise')
+var uuid = require('uuid')
+var crypto = require('crypto')
+var isA = require('joi')
+var error = require('../../lib/error')
+var log = require('../../lib/log')
+
+var TEST_EMAIL = 'foo@gmail.com'
+var MS_ONE_DAY = 1000 * 60 * 60 * 24
+
+var makeRoutes = function (options, requireMocks) {
+  options = options || {}
+
+  var config = options.config || {}
+  config.verifierVersion = config.verifierVersion || 0
+  config.smtp = config.smtp || {}
+  config.memcached = config.memcached || {
+    address: '127.0.0.1:1121',
+    idle: 500,
+    lifetime: 30
+  }
+  config.i18n = {
+    supportedLanguages: ['en'],
+    defaultLanguage: 'en'
+  }
+  config.lastAccessTimeUpdates = {}
+  config.signinConfirmation = config.signinConfirmation || {}
+  config.signinUnblock = config.signinUnblock || {}
+
+  var log = options.log || mocks.mockLog()
+  var Password = options.Password || require('../../lib/crypto/password')(log, config)
+  var db = options.db || mocks.mockDB()
+  var isPreVerified = require('../../lib/preverifier')(error, config)
+  var customs = options.customs || {
+    check: function () {
+      return P.resolve(true)
+    }
+  }
+  var checkPassword = options.checkPassword || require('../../lib/routes/utils/password_check')(log, config, Password, customs, db)
+  var push = options.push || require('../../lib/push')(log, db, {})
+  return proxyquire('../../lib/routes/account', requireMocks || {})(
+    log,
+    require('../../lib/crypto/random'),
+    P,
+    uuid,
+    isA,
+    error,
+    db,
+    options.mailer || {},
+    Password,
+    config,
+    customs,
+    isPreVerified,
+    checkPassword,
+    push,
+    options.devices || require('../../lib/devices')(log, db, push)
+  )
+}
+
+function runTest(route, request, assertions) {
+  return new P(function (resolve) {
+    route.handler(request, function (response) {
+      resolve(response)
```

Comment:
The tests have tried to improve on this error-swallowing pattern, by now doing the following:

```js
if (response instanceof Error) {
  reject(response)
} else {
  resolve(response)
}
```

This helps propagate horrid things like `TypeError` when you typo something into mocha, so it can report the stack trace.